### ISSUE TITLE
some tool names are abbreviated unnecessarily

### ISSUE THREAD

**Issue by andreas-junghanns** [modelica/fmi-crosscheck-tools#20]
... see "FMI Toolbox..."

**Comment by xogeny** [comment]
The full name of that tool is "FMI Toolbox for MATLAB/Simulink".  So abbreviation *is* necessary, right?

**Comment by andreas-junghanns** [comment]
Yes, but only if the box around it is too small. We have seen large boxes and the space not used to show the entire name. Not sure if you have control over that.

**Comment by xogeny** [comment]
Large boxes show up when you zoom in.  There should be no large boxes on the main tool page.  Is this consistent with what you have seen?  If not, I need more details.  If so, how do you want it to behave?

**Comment by andreas-junghanns** [comment]
Attaching screenshot: please check the circles with the same colour - each pair indicating some strangeness
![displaynames](https://user-images.githubusercontent.com/23487124/38421797-ac83dcd6-39a8-11e8-8733-441b37f7d63f.jpg)

**Comment by andreas-junghanns** [comment]
(Note, blue and purple circles show the different ways the tool names are shown. Looks to me like you are not showing the display names in the available and planned tables. this might be w.r.t another ticket...)

W.r.t. abbreviations: I have no idea what you can express - when I tried to understand this a long time ago in the great-grandfather of this iteration of screen-declaration language - so approximately yesterday - I could not get it to work for both constraints. So if you tell me this is still not solved, stop trying to improve.

Here is what think should happen:
 - show the name as long as the box allows
 - minimum box size given
 - make the boxes larger to get as much space as needed

Alternatively, we could make the boxes large enough for the display name and specify a character limit for those display names (I some excess there, seemingly moving the feature list into the name).

Again: If this is still impossible for web, 30 years after tex/latex, we should leave this alone and live with it.

**Comment by sjoelund** [comment]
CSS
```css
  text-overflow: ellipsis;
```
could perhaps work?

**Comment by MartinOtter** [comment]
Not important, but just in case:
I would prefer to only have one tool name per line and then have a longer list of names (seems to me easier to read; with the current display of 4-5 tools per line, it is hard to get quickly an understanding which tools are present). The issue with abbreviated names would then most likely be no longer present.

**Comment by xogeny** [comment]
@andreas-junghanns The issue with the different colored circles is represented by issue #18.  I will investigate other permutations to improve display (but not right now).

@sjoelund I tried `text-overflow: ellipsis` but, IIRC, when used in conjunction with `display: flex` and no max width, it never shrinks the name.  Each name ends up being its full length because there is no constraint to limit the size of the boxes.  Now it might be possible to use something like `max-width` to introduce a constraint so that `text-overflow` was relevant.

@MartinOtter We'd be right back to square one with a list so long that you'd have to scroll through pages to find the tools you were interested in.  Let's not give up on trying to keep data on a single page just yet.

**Comment by MartinOtter** [comment]
Another possibility would be to have two or three columns for the listing, such as:

![image](https://user-images.githubusercontent.com/7248038/38426858-09d59302-39b8-11e8-80d7-c83ad0435a66.png)

I can read much better through alphabetically ordered columns instead of the current "brick" listing with alphabetically ordering in the rows.

**Comment by xogeny** [comment]
@MartinOtter The issue there is making it responsive (i.e., working on different device sizes).  How many columns should there be?  That being said, we could use media queries to determine how many columns.  Another issue is you want them alphabetically vertically...but normally the way CSS flow works is by wrapping so you'd have something more like:

20-sim | ADAMS | Amesim
---- | ---- | ----
**ANSYS Simplorer** | **ASIM-AUTOs** | **AVL CRUISE**


There is also the new "CSS Grid" standard which might be useful here (not sure).

**Comment by MartinOtter** [comment]
I understand your remarks. On the other hand, I am just saying that the current brick listing is unreadable for me. The simplest compromize for me seems still to have only one column (one name per row) and then accept that one might need to scroll (since the time to scroll is much less as to try to read the brick listing).

Yet another option could be to have a button that allows to switch between the two representations (so support both and then the user selects his preferred listing format).

Yes, I understand that a simple solution with CSS is to have all tools names alphabetically sorted and then let CSS insert line breaks depending on the width of the screen. An alternative is to generate an HTML table and then calculate how many columsn are needed, depending on the device width, and if there are, say, 28 tools and 3 columns, then store roundup(28/3)=10 names in the first column, roundup((28-10)/2) = 9 names in the second column and the remaining names in the third column. So this means that some Javascript function is needed to perform these actions.

**Comment by sjoelund** [comment]
> An alternative is to generate an HTML table and then calculate how many columsn are needed [...]

Tables are not supposed to be used for layout though (only tabular data); that's the job of CSS. Anyway, you would need to recalculate the number of columns on a resize event and deal with a lot of the layout issues in JavaScript (the current approach works rather well when you resize the browser). If there is a CSS solution it should be preferred.

**Comment by MartinOtter** [comment]
Yes, I have seen the search button. But, if I want to quickly get an overview which tools are available, then currently the brick layout has to be used (and then I do not get an overview at all, because reading a large brick table is a total mess).

Last try (then I give up):
Look at this [article](https://alistapart.com/article/multicolumnlists) where 6 methods to the "listing of a long list" are described. Method "6" is the preferred solution of the author: _a single semantically-logical list whose column-wrapping is controlled entirely from CSS, relies on normal flow, and works in most modern browsers._ (the result is as I proposed it).

**Comment by MartinOtter** [comment]
Just recognized that this article is from 2005. Another search resulted in the following ready-made solution: https://github.com/yairEO/listBreaker which seems very simple to define and the number of columns depend on the window width (found via https://stackoverflow.com/questions/6509106/is-there-a-way-to-break-a-list-into-columns). Looks as the perfect solution for the fmi crosscheck page.
